cmake/onnxruntime_python.cmake

```diff
@@ -96,11 +70,32 @@ target_link_libraries(onnxruntime_pybind11_state PRIVATE
     ${PROVIDERS_ACL}
     ${PROVIDERS_ARMNN}
     ${PROVIDERS_ROCM}
-    onnxruntime_optimizer
     onnxruntime_providers
     onnxruntime_util
     ${onnxruntime_tvm_libs}
-    onnxruntime_framework
+)
+
+if (onnxruntime_ENABLE_TRAINING)
+  list(APPEND onnxruntime_pybind11_state_link_targets onnxruntime_training)
+  if (onnxruntime_ENABLE_TRAINING_TORCH_INTEROP)
+    list(APPEND onnxruntime_pybind11_state_link_targets onnxruntime_interop_torch)
+  endif()
+endif()
+
+list(APPEND onnxruntime_pybind11_state_link_targets onnxruntime_framework)
+
+if (onnxruntime_ENABLE_TRAINING)
+  if (NOT onnxruntime_ENABLE_TRAINING_TORCH_INTEROP)
+    if (NOT TARGET onnxruntime_python_interface)
+      include(onnxruntime_python_interface.cmake)
+    endif()
+  endif()
+
+  list(APPEND onnxruntime_pybind11_state_link_targets onnxruntime_python_interface)
+endif()
+
+list(APPEND onnxruntime_pybind11_state_link_targets
+    onnxruntime_optimizer
```

Comment:
Doesn't onnxruntime_optimizer depend on onnxruntime_providers?

Reply:
revert the re-ordering.